app/src/main/java/io/apicurio/registry/services/auth/CustomAuthenticationMechanism.java

```diff
@@ -83,20 +95,33 @@ public class CustomAuthenticationMechanism implements HttpAuthenticationMechanis
     @Inject
     AuditLogService auditLog;
 
+    @Inject
+    TenantConfigBean tenantConfigBean;
+
     private BearerAuthenticationMechanism bearerAuth;
 
     private ApicurioHttpClient httpClient;
+    private ApicurioHttpClient resolverHttpClient;
 
     @PostConstruct
     public void init() {
         if (authEnabled) {
+            if (resolveIdentityServer) {
+                resolverHttpClient = new JdkHttpClientProvider().create(resolverRequestBasePath, Collections.emptyMap(), null, new AuthErrorHandler());
+                final IdentityServerResolver.SsoProviders ssoProviders = resolverHttpClient.sendRequest(getSSOProviders(resolverRequestPath));
```

Comment:
These few lines are duplicated again, maybe it's worth it to extract to a single place